python/pysmurf/client/command/smurf_command.py

```diff
@@ -2729,6 +2729,190 @@ def get_jesd_tx_status_valid_cnt(self, bay, num, **kwargs):
+
+        """
+        # strip any commit info off the end of the pysmurf version
+        # string
+        pysmurf_version = self.get_pysmurf_version(**kwargs).split('+')[0]
+
+        # Extra registers allow confirmation of JESD lock for pysmurf
+        # versions >=4.1.0 and Rogue ZIP file versions >=0.3.0.  see
+        # https://github.com/slaclab/pysmurf/issues/467 for more
+        # details.
+        if version.parse(pysmurf_version) >= version.parse('4.1.0'):
+
+            # Will report how long the JESD health check takes to
+            # complete.
+            start_time = time.time()
+
+            # First, check if the 'JesdStatus' register is set to 'Not found'. That means that the
+            # current ZIP file does not contain the new JesdHealth command.
+            status = self.get_jesd_status(**kwargs)
+
+            if status == 'Not found':
+                self.log(
+                    'The `JesdHealth` method is not present in the Rogue'
+                    ' ZIP file.'  , self.LOG_ERROR)
+                return status
+
+            # If the command exists, then start by calling the `CheckJesd`
+            # wrapper command.
+            self._caput(
+                self.epics_root + ':AMCc:SmurfApplication:CheckJesd', 1,
+                wait_done=True, **kwargs)
+
+            # Now let's wait for it to finish.
+            num_retries = int(max_timeout_sec/caget_timeout_sec)
+            success = False
+            status = None
+            for _ in range(num_retries):
+                # Try to read the status register.
+                status = self.get_jesd_status(timeout=caget_timeout_sec, **kwargs)
+
+                if status not in [None, 'Checking']:
+                    success = True
+                    break
+
+            # If after out maximum defined timeout, we weren't able to
+            # read the "JesdStatus" status register with a valid status,
+            # then we exit on error.
+            if not success:
+                self.log(
+                    'JESD health check did not finish after'
+                    f' {max_timeout_sec} seconds.', self.LOG_ERROR)
+                return status
+
+            # Measure how long the process take
+            end_time = time.time()
+
+            self.log(
+                'JESD health check finished after'
+                f' {int(end_time - start_time)} seconds.'
+                f' The final status was {status}.',
+                self.LOG_USER)
+
+            return status
+        else:
+            self.log(
+                'The `SmurfApplication:CheckJesd` and '
+                ' `SmurfApplication:JesdStatus` registers are not'
+                ' implemented for pysmurf core code versions <4.1.0'
+                f' (current version is {pysmurf_version}).',
+                self.LOG_ERROR)
+            return None
+
+    _jesd_status_reg = "JesdStatus"
+
+    def get_jesd_status(self, as_string=True, **kwargs):
```

Comment:
You need to remove the `as_string` argument from this method, due to the comments I added above.

Reply:
removed `as_string` opt args